### DISCUSSION TITLE
how can restart job(rook-ceph-osd-prepare) in rook-ceph v1.8.1 ?

### DISCUSSION THREAD

**Discussion by dong-jae** [rook/rook#9510] [Q&A] [answered]
Rook-Cluster has been installed as version v1.8.1 using helm chart.

In Rook-Ceph v1.7, 
Periodically and automatically, osd-prepare job was relaunched, in v1.8 is not.

i'm trying to [Replace an OSD] by referring to the guide below.
(https://rook.io/docs/rook/v1.8/ceph-osd-mgmt.html)
In order to complete the step, the job restart is essential.

in v1.7, the prepare job can be manually restarted by turning off [rook-ceph-osd-* pod]

Help me, please.

**Comment by leseb** [comment]
What kind of deployment is this? On PVC? Non-PVC? If this is non-PVC the prepare job is always there unless it has been removed. To run it again you can restart the operator.

**Reply by dong-jae** [reply]
@leseb Thank you for the comment.
this cluster running on bare metal (Non-PVC). 
As you comment, I have tried to remove the job and restart the operator, but it didn't work well.

> The guide below is in the document.
>> 4. The operator ideally will automatically create the new OSD within a few minutes of adding the new device or updating the CR. If you don’t see a new OSD automatically created, restart the operator (by deleting the operator pod) to trigger the OSD creation.

It worked well in v1.7, but the same way doesn't work well in v1.8.
(a raw device ready to be replaced as a new osd is already there.)

**Reply by leseb** [reply]
Please share the operator's log, this shouldn't be happening.

**Reply by dong-jae** [reply]
@leseb i have tried again and log file's here (the log file saved 30 minutes passed restarting operator.)
[rook-ceph-operator-5c479f84d7-6l2lm.log](https://github.com/rook/rook/files/7811683/rook-ceph-operator-5c479f84d7-6l2lm.log)
 
target host is "pbla-k8slb-t1802" and completely <osd.1> removed. 

And Raw partitions (no formatted filesystem) for new osd is ready. (/dev/sda3)

Tell me if you need more logs.
------ osd tree
```
ID   CLASS  WEIGHT   TYPE NAME                   STATUS     REWEIGHT  PRI-AFF
 -1         0.26093  root default                                            
 -3         0.05219      host nbla-kfkdev-t1801                              
  0    hdd  0.05219          osd.0                      up   1.00000  1.00000
 -7         0.05219      host nbla-kfkdev-t1802                              
  2    hdd  0.05219          osd.2                      up   1.00000  1.00000
 -9         0.05219      host nbla-kfkdev-t1803                              
  3    hdd  0.05219          osd.3                      up   1.00000  1.00000
-11         0.05219      host pbla-k8slb-t1801                               
  4    hdd  0.05219          osd.4                      up   1.00000  1.00000
 -5         0.05219      host pbla-k8slb-t1802                               
  1    hdd  0.05219          osd.1               destroyed         0  1.00000
```

**Comment by travisn** [comment, marked as answer]
The log shows the error: 
```
2022-01-05 02:28:26.942491 E | ceph-cluster-controller: failed to reconcile CephCluster "rook-ceph/rook-ceph". failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph mgr: failed to enable mgr services: 
failed to enable service monitor: service monitor could not be enabled: failed to retrieve servicemonitor. servicemonitors.monitoring.coreos.com "rook-ceph-mgr" is forbidden: 
User "system:serviceaccount:rook-ceph:rook-ceph-system" cannot get resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "rook-ceph"
```

Until this is resolved, the operator will not create OSDs. Did you create the [rbac.yaml](https://rook.github.io/docs/rook/v1.8/ceph-monitoring.html) required to enable monitoring?

**Reply by dong-jae** [reply]
@leseb @travisn, Thank you so much for your help.

i manually have created rbac.yaml. 
```
role.rbac.authorization.k8s.io/rook-ceph-monitor created
rolebinding.rbac.authorization.k8s.io/rook-ceph-monitor created
role.rbac.authorization.k8s.io/rook-ceph-metrics created
rolebinding.rbac.authorization.k8s.io/rook-ceph-metrics created
role.rbac.authorization.k8s.io/rook-ceph-monitor-mgr created
rolebinding.rbac.authorization.k8s.io/rook-ceph-monitor-mgr created
```
And then now works well perfectly!

On the Helm chart,i set monitoring.enabled to true.
i think if monitoring is enabled, automatically will create RBAC. but didn't.
did I misunderstand that?
(my cluster's helm chart version is v1.8.1)
```
monitoring:
  # requires Prometheus to be pre-installed
  # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
  enabled: true
```

**Reply by travisn** [reply]
Good to hear it's working, do you want to open a PR for that suggestion?

**Reply by dong-jae** [reply]
@travisn Sorry, I'm late to check. 
i'd like to, but it was my mistake.
i have missed Ceph Operator's "values.yaml".

up to helm chart v1.8.0, in Ceph Operator's values.yaml didn't contain monitoring.enabled 
and monitoring.enabled only exists Ceph Cluster's values.yaml.

i am using v.1.8.1, monitoring.enabled have set only on Ceph Cluster's values.yaml.
i added monitoring.enabled(true) to Ceph Operator's values.yaml too, now RBAC create works well.

Although the configuration of the Helmchart template is a little confusing, it has been solved well anyway.
